pkg/network/master/vnids.go

```diff
@@ -77,26 +75,17 @@ func (vmap *masterVNIDMap) isAdminNamespace(nsName string) bool {
 	return false
 }
 
-func (vmap *masterVNIDMap) populateVNIDs(netNamespaceInformer networkinformers.NetNamespaceInformer) error {
```

Comment:
This patch doesn't work. You're removing the initial bulk NetNamespace->netIDManager initialization, but there isn't any lazy NetNamespace->netIDManager initialization anywhere, so it will never learn the existing namespace->VNID mappings. So when you start up the master the second time, it will see Namespace Added events for each pre-existing Namespace, and it will call assignVNID, which will call allocateNetID, which will call getVNID to see if the Namespace is already known, but it's not already known, so that will return !found, and so then allocateNetID will call vmap.netIDManager.AllocateNext() to allocate a new VNID, then try to Create a NetNamespace object with that VNID, fail because the NetNamespace already exists, and then log an error.

Even if you updated handleAddOrUpdateNetNamespace to do the right thing, you'd still have to deal with the fact that you don't know if you're going to see Namespace "foo" or NetNamespace "foo" first. So I think basically you can't get rid of populateVNIDs.

Reply:
fixed, initializing vnid allocator is done in https://github.com/openshift/origin/pull/18911/commits/00bf4a79a7d6f33c32fb141a299ff94655bf5245